spring-integration-scripting/src/main/java/org/springframework/integration/scripting/jsr223/ScriptExecutorFactory.java

```diff
@@ -31,7 +39,32 @@ public static ScriptExecutor getScriptExecutor(String language) {
 		else if (language.equalsIgnoreCase("ruby") ||  language.equalsIgnoreCase("jruby")) {
 			return new RubyScriptExecutor();
 		}
+		else if (language.equalsIgnoreCase("kotlin")) {
+			return new KotlinScriptExecutor();
+		}
 		return new DefaultScriptExecutor(language);
 	}
 
+	/**
+	 * Derive a scripting language from the provided script file name.
+	 * @param scriptLocation the script file to consult for extension.
+	 * @return the language name for the {@link ScriptExecutor}.
+	 * @since 5.2
+	 */
+	public static String deriveLanguageFromFileExtension(String scriptLocation) {
+		int index = scriptLocation.lastIndexOf(".") + 1;
+		Assert.state(index > 0, () -> "Unable to determine language for script '" + scriptLocation + "'");
+		String extension = scriptLocation.substring(index);
+		if (extension.equals("kts")) {
+			return "kotlin";
+		}
+		ScriptEngineManager engineManager = new ScriptEngineManager();
+		ScriptEngine engine = engineManager.getEngineByExtension(extension);
+		Assert.state(engine != null, () -> "No suitable scripting engine found for extension '" + extension + "'");
+		return engine.getFactory().getLanguageName();
+	}
+
+	private ScriptExecutorFactory() {
```

Comment:
> Needs a private CTOR

Doesn't this work?

Reply:
Sorry; missed that 😦

Reply:
But it needs `super()` to prevent Sonar from complaining.

Reply:
OK. Will add. Can we tweak Sonar do not complain on the matter?
There is definitely a `super()` in the target byte code, so it's complaining is pointless.

Or do I miss something else ?

Reply:
Adding super is a no-op - the compiler generates one anyway.

I don't think we want to disable empty block detection.